File: odoo-portal-fr/app/lib/risk.ts
export type InvoiceRiskResponse = {
  late_payment_risk: number;
  risk_level: "LOW" | "MEDIUM" | "HIGH";
};

export async function getInvoiceRisk(
  amountTotal: number,
  expectedDays: number
): Promise<InvoiceRiskResponse> {
  const token = localStorage.getItem("token");
  if (!token) throw new Error("Not authenticated");

  const res = await fetch(
    `http://127.0.0.1:9000/risk/invoice?amount_total=${amountTotal}&expected_days=${expectedDays}`,
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  if (!res.ok) {
    throw new Error("Failed to fetch invoice risk");
  }

  return res.json();
}

File: api_server/routers/risk.py
from fastapi import APIRouter, Query
from ml.predict import predict_late_risk

router = APIRouter(prefix="/risk", tags=["Risk"])

@router.get("/invoice")
def invoice_risk(
    amount_total: float = Query(..., gt=0),
    expected_days: int = Query(..., gt=0)
):
    risk = predict_late_risk(amount_total, expected_days)

    if risk > 0.7:
        level = "HIGH"
    elif risk > 0.4:
        level = "MEDIUM"
    else:
        level = "LOW"

    return {
        "late_payment_risk": round(risk, 3),
        "risk_level": level
    }
